src/store/store.tsx

```diff
@@ -74,6 +75,39 @@ export async function studyStoreCreator(
       setIframeProvenance: (state, action: PayloadAction<TrrackedProvenance | null>) => {
         state.iframeProvenance = action.payload;
       },
+      setMatrixAnswersRadio: (state, action: PayloadAction<{ questionKey: string, responseId: string, val: string }>) => {
+        const { responseId, questionKey, val } = action.payload;
+        return {
+          ...state,
+          matrixAnswers: {
+            ...state.matrixAnswers,
+            [responseId]: {
+              ...state.matrixAnswers[responseId],
+              [questionKey]: val,
+            },
+          },
+        };
+      },
```

Comment:
Generally, redux is supposed to return a new object because it replaces the whole object -- not just set particular values. This is meant to ensure immutability. We shouldn't actually be mutating the state directly like the other pieces.